tests/span_tests.cpp

```diff
@@ -110,17 +105,17 @@ TEST(span_test, from_nullptr_size_constructor)
         std::abort();
     });
     {
-        span<int> s{nullptr, narrow_cast<span<int>::index_type>(0)};
+        span<int> s{nullptr, narrow_cast<span<int>::size_type>(0)};
```

Comment:
can we get rid of the narrow_cast here?

Reply:
the cast itself will have to stay, but maybe we should just use static_cast instead of narrow_cast.

Reply:
for clarity, I'm going to leave the narrow_cast